packages/gafl-webapp-service/src/pages/contact/address/entry/route.js

```diff
@@ -5,13 +5,13 @@ import { validation } from '@defra-fish/business-rules-lib'
 import { referenceDataOperations } from '../../../../services/sales-api/sales-api-service.js'
 
 const validator = Joi.object({
-  premises: validation.contact.premisesValidator,
-  street: validation.contact.streetValidator,
-  locality: validation.contact.localityValidator,
-  town: validation.contact.townValidator,
+  premises: validation.contact.createPremisesValidator(Joi),
```

Comment:
Did we really get to the bottom of this problem? I don't think I experienced the issue in the front end, so the question still remains. I think we need to find out why. This work-around is Ok but it is rather cumbersome.

Reply:
Not entirely - it appears that Joi builds an internal list of validators and as we were previously not passing in the global Joi object it was returning the wrong validation error in the API.  I proved this by copying the shared validators into the SalesAPI project where everything immediately started working correctly but referencing via the business-rules-lib project failed.  I don't think it is worth the time to debug deep into the bowels of Joi to find the root cause.  I suspect if you were using nested/hierarchical structures in the frontend then you would've encountered this problem also.